toolkit/themes/shared/design-system/config.js

```diff
@@ -52,23 +52,57 @@ let customFileHeader = surface => {
     " */",
   ].join("\n");
 
+  // let layerString = ["@layer tokens-foundation, tokens-prefers-contrast, tokens-forced-colors;"];
+
   let cssImport = surface
     ? `@import url("chrome://global/skin/design-system/tokens-shared.css");\n\n`
     : "";
-
-  return [licenseString + "\n\n" + commentString + "\n\n" + cssImport];
+  let layerString = !surface ? `@layer tokens-foundation, tokens-prefers-contrast, tokens-forced-colors;\n\n` : "";
+
+  return [
+    licenseString + "\n\n" + 
+    commentString + "\n\n"  +
+    cssImport +
+    layerString
+  ];
 };
 
+const NEST_MEDIA_QUERIES_COMMENT = `/* Bug 1879900: Can't nest media queries inside of :host, :root selector
+   until Bug 1879349 lands */`;
+
 const MEDIA_QUERY_PROPERTY_MAP = {
   "forced-colors": "forcedColors",
   "prefers-contrast": "prefersContrast",
 };
 
-const BASE_SELECTOR = ":root,\n" + ":host(.anonymous-content-host) {\n";
+const TOKENS_LAYERS_MAP = {
+  "default": "tokens-foundation",
+  "prefersContrast": "tokens-prefers-contrast",
+  "forcedColors": "tokens-forced-colors",
+};
 
 function formatBaseTokenNames(str) {
   return str.replaceAll(/(?<tokenName>\w+)-base(?=\b)/g, "$<tokenName>");
 }
+/**
+ * Generates the necessary selectors and structure for our layers and media queries
+ *
+ * @param {number} numberOfSpaces Used to control the level of indentation
+ * @param {string} formattedVars The output of the formatVariables function
+ * @param {string} layer One of the keys from TOKENS_LAYERS_MAP, 
+ *  used to control the closing indentation level
+ * @return {string} A formatted string that contains the layer selector,
+ *  base (:root, :host(.anonymous-content-host)) selector,
+ *  media query (if present), and the formatted CSS variables
+ */
+function generateBaseSelector(numberOfSpaces, formattedVars, layer) {
+  let indentation = " ".repeat(numberOfSpaces);
+  // More weird spacing to keep the indentation correct
+  return `${indentation}:root,
+${indentation}:host(.anonymous-content-host) {
+${formattedVars}
+${layer != "default" ? indentation : "  "}}`;
+}
```

Comment:
I'm not too happy with this function, so if someone has a better suggestion I am all ears